File: nightly/tests/python/globalflow.py
# Taint through a module-level global. produce() writes source() into g_data;
# consume() reads it back and sinks it. Neither function passes anything to the
# other, so the flow exists only if taint is tracked through global storage
# across calls.
g_data = None


def produce():
    global g_data
    g_data = source()


def consume():
    sink(g_data)


def main():
    produce()
    consume()

File: nightly/tests/c/globalflow.c
/* Taint through a file-scope global, the C analogue of
 * tests/java/StaticFieldFlow.java and CrossClassStaticFieldFlow.java:
 *   produce() writes source() into the global
 *   consume() reads the global and passes it to the sink
 * Neither function passes anything to the other, so the flow exists only if
 * the analysis tracks taint through the global's storage across calls.
 */

int source(void);
void sink(int x);

int g_data;

void produce(void) {
  g_data = source();
}

void consume(void) {
  sink(g_data);
}

int main(void) {
  produce();
  consume();
  return 0;
}
